code/datums/uplink_items/uplink_traitor.dm

```diff
@@ -587,33 +593,33 @@
 	U.used_TC = cost
 
 	var/datum/uplink_item/I
-	var/danger_counter = 0 // lets make sure we dont get into an infinite loop...
 	while(remaining_TC)
-		if(danger_counter > RECURSION_PANIC_AMOUNT)
-			fucked_shit_up_alert(loc, "[src] spawning failed: approached an infinite loop by cost checking, giving the remaining [remaining_TC] telecrystals instead.")
-			generate_refund(remaining_TC, C)
+		if(remaining_TC < LEGACY_TC)
+			generate_refund(cost, C)
+			message_admins("[src] has refunded [remaining_TC] because it was less than [LEGACY_TC]. [ADMIN_COORDJMP(loc)]")
```

Comment:
```suggestion
			generate_refund(remaining_TC, C)
			message_admins("[src] has been given [remaining_TC] because it was less than [LEGACY_TC] worth in their [name]. [ADMIN_COORDJMP(get_turf(loc))]")
```

Reply:
Guh I thought I fixed the cost, the rest was an oversight on my part

Reply:
Don't do this, this will not work as an embed